### PR TITLE
[testgridshot] upload images to a bucket

### PR DESCRIPTION
**Pull request by hoegaarden**
As a followup to https://github.com/kubernetes/release/pull/882#issuecomment-551328102 we could store the images in a google storage bucket. This would supersede #882.

That would be relatively simple, assuming the person who calls the script is a a release-manager and therefore should have access to certain buckets and the gcloud tools setup correctly already. So no more special setup if my assumptions are correct.

I am not sure, to which bucket we'd want to push the images too. Should we create a new one, should we use an already existing one?

cc: @kubernetes/release-engineering 
Closes: #882 
/kind cleanup

### CODE REVIEW

**Comment by hoegaarden** [comment]
> Let me give some thought to what we call the bucket before I create it.

 I think I'd slightly lean towards reusing an existing bucket, mostly because I don't have a good overview of all the buckets and their intended use, and don't want to introduce yet another one.
You probably have a better overview on all of that. Can you maybe give your reasoning why you'd prefer a new bucket over using an existing one, e.g. `kubernetes-release` or such?

**Comment by hoegaarden** [comment]
Example bucket layout after c32934e00:

```console
→ gsutil ls -lr gs://$BUCKET/
gs://$BUCKET/testgridshot/:
        11  2019-11-26T12:19:48Z  gs://$BUCKET/testgridshot/

gs://$BUCKET/testgridshot/1.15/:

gs://$BUCKET/testgridshot/1.15/2019-11-27_10-18-42_0f1c8d9a/:
   1677303  2019-11-27T10:18:51Z  gs://$BUCKET/testgridshot/1.15/2019-11-27_10-18-42_0f1c8d9a/sig_release_1_15_informing_gce_windows_stable1.jpg

gs://$BUCKET/testgridshot/1.16/:

gs://$BUCKET/testgridshot/1.16/2019-11-27_10-18-56_c485091f/:
   1611709  2019-11-27T10:19:04Z  gs://$BUCKET/testgridshot/1.16/2019-11-27_10-18-56_c485091f/sig_release_1_16_informing_gce_windows_beta.jpg

gs://$BUCKET/testgridshot/1.17/:

gs://$BUCKET/testgridshot/1.17/2019-11-27_09-58-32_7eba5288/:
   1182460  2019-11-27T09:58:46Z  gs://$BUCKET/testgridshot/1.17/2019-11-27_09-58-32_7eba5288/sig_release_1_17_informing_aks_engine_azure_1_17_windows.jpg
   1627746  2019-11-27T09:58:46Z  gs://$BUCKET/testgridshot/1.17/2019-11-27_09-58-32_7eba5288/sig_release_1_17_informing_gce_cos_k8sbeta_default.jpg
   1584127  2019-11-27T09:58:46Z  gs://$BUCKET/testgridshot/1.17/2019-11-27_09-58-32_7eba5288/sig_release_1_17_informing_gce_windows_1_17.jpg

gs://$BUCKET/testgridshot/master/:

gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/:
   1204008  2019-11-27T10:03:20Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/sig_release_master_informing_aks_engine_azure_1_17_windows.jpg
   1530266  2019-11-27T10:03:20Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/sig_release_master_informing_capa_conformance_stable_k8s_master.jpg
   1582567  2019-11-27T10:03:20Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/sig_release_master_informing_capg_conformance_stable_k8s_master.jpg
    309229  2019-11-27T10:03:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/sig_release_master_informing_gce_master_scale_performance.jpg
   1654753  2019-11-27T10:03:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-03-00_77288065/sig_release_master_informing_gce_windows_master.jpg

gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/:
   1203871  2019-11-27T10:19:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/sig_release_master_informing_aks_engine_azure_1_17_windows.jpg
   1530212  2019-11-27T10:19:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/sig_release_master_informing_capa_conformance_stable_k8s_master.jpg
   1582281  2019-11-27T10:19:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/sig_release_master_informing_capg_conformance_stable_k8s_master.jpg
    309279  2019-11-27T10:19:19Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/sig_release_master_informing_gce_master_scale_performance.jpg
   1654590  2019-11-27T10:19:20Z  gs://$BUCKET/testgridshot/master/2019-11-27_10-19-08_a515ae9d/sig_release_master_informing_gce_windows_master.jpg
TOTAL: 16 objects, 20244412 bytes (19.31 MiB)

→ 
```

File: testgridshot [hunk 1]
```diff
@@ -28,17 +28,18 @@
...
-#+      (https://vgy.me/auth/register) and a user key to be able to upload images.
-#+
 #+   BOARDS: [default: "blocking informing"]
 #+     Change which boards you are interested in
 #+
 #+   STATES: [default: "FAILING"]
 #+     Change the tests you want to screenshot
 #+     Available states: FAILING, FLAKY, PASSING
 #+
+#+   BUCKET: [default: "hhorl-test"]  # TODO(hoegaarden)
```
Comment (justaugustus) [inline testgridshot]:
```suggestion
#+   BUCKET: [default: "kubernetes-release-gcb"]
```

File: testgridshot [hunk 2]
```diff
@@ -80,6 +76,7 @@ set -o pipefail
 readonly BOARDS="${BOARDS:-blocking informing}"
 # Available states: FAILING, FLAKY, PASSING
 readonly STATES="${STATES:-FAILING}"
+readonly BUCKET="${BUCKET:-hhorl-test}"  # TODO(hoegaarden)
```
Comment (justaugustus) [inline testgridshot]:
```suggestion
readonly BUCKET="${BUCKET:-kubernetes-release-gcb}"
```

**Comment by hoegaarden** [comment]
The default bucket has been changed to `kubernetes-release-gcb`.

**Comment by justaugustus** [comment]
Thanks Hannes!
/lgtm
/approve
/milestone v1.17

**Pull request merged**